### PR TITLE
Adds IvoryTransport

### PR DESCRIPTION
**Pull request by sagikazarmark**
I required a development version of the adapter package. The author promised to do some work on it on this weekend. Until a new version is tagged here, I don't think it is a problem.

### CODE REVIEW

**Comment by lstrojny** [comment]
Looks good. Can you add that one to the integration test as well?

**Comment by sagikazarmark** [comment]
Of course, but as I can see, the tests would still fail. Anyway, I still think it would be a lot more easier to just remove the other transports.

File: src/fXmlRpc/Transport/IvoryTransport.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ */
+namespace fXmlRpc\Transport;
+
+use fXmlRpc\Exception\TcpException;
+use fXmlRpc\Exception\HttpException;
+use Ivory\HttpAdapter\HttpAdapterException;
+use Ivory\HttpAdapter\HttpAdapterInterface;
+use Ivory\HttpAdapter\Message\Response;
+
+final class IvoryTransport implements TransportInterface
```
Comment (lstrojny) [inline src/fXmlRpc/Transport/IvoryTransport.php]: Should implement HttpTransportInterface

Comment (sagikazarmark) [inline src/fXmlRpc/Transport/IvoryTransport.php]: I was also thinking about that, and I chose not to implement it. The reason is that HttpTransportInterface is some sort of builder to be able to set custom data/headers. Ivory has built-in support for it.

Comment (lstrojny) [inline src/fXmlRpc/Transport/IvoryTransport.php]: Understood. Without knowing too much about the details of Ivory, how hard would it be to support it?

Comment (sagikazarmark) [inline src/fXmlRpc/Transport/IvoryTransport.php]: I don't think it is too hard. It provides a great interface for HTTP requests, implements the upcoming PSR7 interface. I use it in several places, also a few well-known packages uses/will use it (eg. Geocoder 3.0). The author replies to all issues, is open to ideas.

**Comment by lstrojny** [comment]
Closing in favor of #26

**Pull request closed without merging**